### ISSUE TITLE
API Implementation

### ISSUE THREAD

**Issue by Matroxt** [zedd3v/abck-dev#1]
Hello there!

I was wondering if there were any interest in developing an API for your parser.

### Parsing
Method: POST
Endpoint: `/api/parse`
Intake Payload: 
```json
{"sensor": "<bmak_here>"}
```
Output Payload: 
```json
{ 
    "bmak.xagg": "val", 
    "bmak.lang": "val", 
    "etc": "..." 
}
```
### Comparing
Method: POST
Endpoint: `/api/compare`
Intake Payload: 
```json
{ "sensor1": "<bmak_here>", "sensor2": "<bmak_here>" }
```
Output Payload (where only differences are outputed): 
```json
{
    "bmak.xagg": {
        "sensor1": "val", 
        "sensor2": "diffval"
    }
}
```

## Reasoning

A couple weeks back, I made essentially the same parser for BMAK data but in Go and as an API instead (based on the older version of the website)
It takes a unparsed BMAK data as input payload, and outputs the value in a formatted JSON.

Now, my parser seems already a bit out of date because Akamai updates their versions quite often and some fields don't work well anymore.

What would make more sense is for the community to have this unique code base that meets all their need for BMAK parsing IMO.

## Possible Usecase
One could export the JSON to services like Splunk, Logstash/Kibana etc and make data analysis on what BMAK work and what doesn't. Allowing for quicker identification of problems and data driven action.

## Caveats
Of course, we'd need a config file to enable/disable that feature because having such endpoint on the public internet would cause too much load on the website, but it could be really neat if people deployed their own API for their own use.

## TLDR
Would you have any interest in adding an API endpoint to this code base?
If you don't want to invest time in this, I can give it a try.
I myself don't have much experience with TS/React but given enough time and the assurance that you would merge functioning code, I'd probably do it.

Let me know what you think! 😃

**Comment by zedd3v** [comment]
https://abck.dev/api

done

**Comment by Matroxt** [comment]
Just to clarify:
I see you added documentation called API without actually adding code for any API [in this commit]( https://github.com/zedd3v/abck-dev/commit/665f7ae1528ee2598025ef12da9b48ac867913be).
Are you gonna add that code later or is this a go for me to start working on it?

**Comment by zedd3v** [comment]
the api is live already, i just didnt release the code hosting on my private vps

**Comment by Matroxt** [comment]
Any plans to make it open source? 

I got multiple thousands of different sensor data I'd like to parse per day, I wouldn't want to use too many resources on your VPS. (Or being reliant on a 3rd party for something I'd like to have 100% uptime if I'm being honest)

Ideally, I'd like to make a Dockerfile for your app and deploy it on my k8s cluster

**Comment by zedd3v** [comment]
it can handle atleast 10k requests per second

**Comment by Matroxt** [comment]
i believe you on that one, no worries 😂 

My point is that if I integrate the parsing of sensor via API in my flow, I need to make sure it'll still be up for the duration I need it (months? years?).
I can't be sure yours will be up by then which is why I'd prefer an open source version I can host on my infra.
